sdk/storage/azure-storage-common/src/crypt.cpp

```diff
@@ -16,11 +16,59 @@
 #endif
 
 #include <stdexcept>
+#include <vector>
 
 #include "azure/storage/common/storage_common.hpp"
 
 namespace Azure { namespace Storage {
 
+  namespace Details {
+    std::string UrlEncodeQueryParameter(const std::string& value)
```

Comment:
Why are we re-implementing this. Can you use the `Encode` method for this?
https://github.com/Azure/azure-sdk-for-cpp/blob/b6c1da62c6da40ee1bdd10611b9517198bb0b54d/sdk/core/azure-core/inc/azure/core/http/http.hpp#L199-L210